scripts/pyproject.toml

```diff
@@ -19,7 +19,6 @@ ignore = [
   "T",     # flake8-print
   "FBT",   # flake8-boolean-trap
   "PERF",  # perflint
-  "ANN401",
```

Comment:
Refactoring the implementation as described [here](https://github.com/astral-sh/ruff/pull/5601#discussion_r1260543950) introduced a subtle bug which this PR is solving but the bug exists in the parent PR. Ideally, the stack should've been in the reverse order such that this PR is based on the `main` branch but I didn't realize this until quite late. Then, the rebase required a lot of effort which didn't seem worthwhile.

So, the simple solution is to disable the check on the parent PR and enable it back again in this PR. Sorry for the confusion!